docs/Developers/Linter_Bears_Advanced.rst

```diff
@@ -41,6 +40,30 @@ The path of the temporary configuration file can be accessed inside
         def create_arguments(filename, file, config_file):
             return "--use-config", config_file
 
+If you return ``None``, no configuration file is generated. A common case
+where to explicitly return ``None`` is when you want to expose a setting
+for the user to use his own tool-specific config. In case the user specifies
+such a config file, we can avoid generating one again with ``generate_config``
+to reduce I/O load.
+
+::
+
+    @linter(executable='...')
+    class MyBear:
+        @staticmethod
+        def generate_config(filename, file,
+                            user_config: str = '',
+                            setting_a: str = ''):
```

Comment:
Optional: Do you want to make it bool? As below there's `A=True`. But the type can be arbitrary.

Reply:
I think making it `bool` will be better.